### PR TITLE
Add number answer type

### PR DESCRIPTION
**Pull request by DavidBiddle**
#### What problem does the pull request solve?
Adds a new number input type. This can be any whole or decimal number greater than or equal to 0.

Trello card: https://trello.com/c/tyDQQZ6A/222-add-number-answer-to-a-question-in-the-runner-and-api

#### Checklist

- [x] I've used the pull request template
- [x] I've linked this PR to the relevant issue (if mission work)
- [x] I've written unit tests for these changes (if code change)
- [n/a] I've updated the documentation in (If any documentation requires updating)
    - [n/a] README.md
    - [n/a] Elsewhere (please link)

### CODE REVIEW

File: spec/services/notify_service_spec.rb [hunk 1]
```diff
@@ -22,7 +22,7 @@
 
         travel_to submission_datetime do
```
Comment (DavidBiddle) [inline spec/services/notify_service_spec.rb:23]: (All the changes in this file are automatic Rubocop formatting changes)

File: spec/requests/base_spec.rb [hunk 2]
```diff
@@ -280,10 +280,6 @@
           it "Render the not found page" do
             expect(response.body).to include(I18n.t("not_found.title"))
           end
-
```
Comment (DavidBiddle) [inline spec/requests/base_spec.rb:283]: Rubocop suggested this (it's a duplicate test - there's an identical one earlier in the file that's not shown in this preview)

File: config/locales/en.yml [hunk 3]
```diff
@@ -33,6 +33,12 @@ en:
             national_insurance_number:
               blank: Enter a National Insurance number
               invalid_national_insurance_number: Enter a National Insurance number that is 2 letters, 6 numbers, then A, B, C or D, like QQ 12 34 56 C
+        question/number:
+          attributes:
+            number:
+              blank: Enter a number
+              greater_than_or_equal_to: The answer must be greater than or equal to 0
```
Comment (danworth) [inline config/locales/en.yml:40]: At risk of being pedantic but maybe `greater_than_or_equal_to_zero`?

**Review by danworth — APPROVED** [review]
I've been trying this out locally in addition to the tests included and it looks good to me. I haven't looked at what comes out of Notify but if there's no reason to suspect this adds anything to what @aliuk2012 is already working on then I think that's ok.

I noticed that we keep the leading 0s such as `0004` which I guess is the desired outcome? Also it allows `-0` which seems fine and I was just trying things out, led me down a rabbit hole of wondering if negative zero is even a thing, turns out it might be: https://en.wikipedia.org/wiki/Signed_zero#:~:text=The%20notation%20%22%E2%88%920%22%20may,statistical%20mechanics%20and%20other%20disciplines.

File: config/locales/en.yml (hunk 3 above)
Comment (DavidBiddle) [reply config/locales/en.yml:40]: (but you're right, that would be better)

Comment (danworth) [reply config/locales/en.yml:40]: Ah cool, didn't know it was auto generated 👍

File: spec/models/question/number_spec.rb [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+RSpec.describe Question::Number, type: :model do
+  it_behaves_like "a question model"
+
+  context "when given an empty string or nil" do
+    it "returns invalid with blank message" do
+      question = described_class.new
+      question.validate
+      expect(question.errors[:number]).to include(I18n.t("activemodel.errors.models.question/number.attributes.number.blank"))
+      question.number = ""
+      question.validate
```
Comment (aliuk2012) [inline spec/models/question/number_spec.rb:13]: We can simplify this ;-)

**Pull request merged**